drivers/gpu/drm/sunxi/sunxi_drm_plane.c

```diff
@@ -573,7 +574,8 @@ static int sunxi_plane_mode_set(struct disp_layer_config_data *cfg, struct drm_c
 	config->info.screen_win.width = crtc_w;
 	config->info.screen_win.height = crtc_h;
 	/* 3D mode set in property */
-	config->info.alpha_mode = 1;
+    /* Hack? Set alpha mask to per-pixel for cursor planes*/
+	config->info.alpha_mode = config->info.zorder > 0  ? 0 : 1;
```

Comment:
I see no reason why this cannot be a branchless assignment:

```c
config->info.alpha_mode = config->info.zorder <= 0;
```

Reply:
I prefer explicitness here.